onnxruntime/contrib_ops/rocm/bert/attention_softmax.h

```diff
@@ -122,21 +122,16 @@ __device__ inline void SoftmaxSmall(const int all_sequence_length,
 
   bool is_valid = false;  // whether it has attention mask == 1.
 
-  // Update end position for unidirectional.
+  // Update end position for causal.
   int end = valid_end;
-  if (is_unidirectional) {
-    int end_unid = all_sequence_length - sequence_length + (blockIdx.x % sequence_length) + 1;
-    if (end_unid <= valid_start) {
-      // In this situation, mask of [0, end_unid) and [valid_start, valid_end) has -10000,
-      //              and [end_unid, valid_start) and [valid_end, all_seq_len) has -20000.
-      // So [0, end_unid) will also have value after softmax.
-      is_valid = threadIdx.x < end_unid;
-    } else {
-      end = min(valid_end, end_unid);
```

Comment:
The old code was for mask filter value of -10000. It does not have parity with huggingface transformers latest code with mask filter value of lowest float.